### DISCUSSION TITLE
Payment currency different to base vs reporting and future viewing of data

### DISCUSSION THREAD

**Discussion by rob-c-baker** [craftcms/commerce#2495] [Ideas] [answered]
If Commerce considers other currencies based on conversion rates from the "base" currency, how does it deal with viewing and reporting on orders that were placed in the past (i.e. after the conversion rate has been updated for a non-base currency). I'm guessing by use of the `|commereceCurrency` Twig filter, it only uses the current conversion rate, thus causing a potential inaccuracy in amounts. Is that right?

Thinking on, while using the Stripe payment integration, the Stripe control panel shows payments in the currency the user chose on the Craft site at the time of purchase. The "Amount" (base currency) and "Payment Amount" (payment currency) are shown separated in Commerce's order screen under the "Transactions" tab. Looking at the database, these are stored as 2 values, with the corresponding currency codes. This should mean that if we report based on that table, it should be more accurate. Right?

If I'm right above and we report that way, it solves the accountants getting upset that the numbers are not right but we would still have the situation where any order notfication emails / screens / viewing of the data *might* be inaccurate if the currency rate changed between placement of order and viewing the data again, via whatever mechanism.

I think I read elsewhere that true multi-currency for Commerce is coming, but are there any ideas how to mitigate this sort of thing until then?

**Comment by rob-c-baker** [comment]
One thought I had was that the current `orders` db table has a `paymentCurrency` column. If it also has a `paymentCurrencyConversionRate` _(needs a better name!)_ column then that could be used in the `|commereceCurrency` Twig filter somehow, producing more accurate results.

**Comment by lukeholder** [comment, marked as answer]
All transactions on the orders store the `amount` (in store currency), `paymentAmount` (in payment currency), `currency` (the store currency code), `paymentCurrency` (payment currency code), `paymentRate` the snapshot of the conversion rate (e.g `1.2`) used between the `amount` and the `paymentAmount`. 

The order will always have its amounts/totals in the store currency. total paid on the order is the total of the successful transactions on the order (sum of `transaction.amount`). If you wanted to report on the currency used to pay, and what the rate used was, you would need to included the transactions in your reporting.

**Reply by lukeholder** [reply]
Also note that we support partial payments (multiple payments) on an order, so in theory the conversion rate may be different between the 2 transactions due to currency flux, and so we can't store it on the order as a single conversion rate number.

The `order.paymentCurrency` is just the last currency used to make payment.

**Reply by rob-c-baker** [reply]
OK, I see. I should be able to get closer to what we need by using the `paymentRate` field - I had missed that column. If not, I'll use the transaction table too.

Thanks for looking at this.
